src/Legacy/Storage.php

```diff
@@ -401,10 +401,10 @@ public function saveContent(Content $content, $comment = null)
         }
 
         // Test to see if this is a new record, or an update
-        if (empty($fieldvalues['id'])) {
-            $create = true;
-        } else {
-            $create = false;
+        $create = empty($fieldvalues['id']);
+
+        if (!isset($fieldvalues['slug'])) {
+            $fieldvalues['slug'] = ''; // Prevent 'slug may not be NULL'
```

Comment:
…and apologies for the initial spare response, I was on my way out the door at the time